src/utils/session.ts

```diff
@@ -42,8 +40,24 @@ export function joinRoom() {
     return;
   }
   peer.value?.destroy();
-  peer.value = new Peer(PEER_JS_OPTIONS);
-  console.info(`Join room ${roomName} as ${playerName}.`);
+  peer.value = new Peer(hostId.value, PEER_JS_OPTIONS);
+  const errorHandler = (e: PeerError<string>) => {
+    switch (e.type) {
+      case "unavailable-id":
+        peer.value?.destroy();
+        peer.value = new Peer(PEER_JS_OPTIONS);
+        console.info(`Join room ${roomName} as ${playerName}.`);
+        break;
+    }
+  };
+  const openHandler = () => {
+    console.info(
+      `Create a room with name ${roomName} as ${playerName} and wait for players to join.`
+    );
+    playerMap.value = new Map([[peer.value.id, playerName.value]]);
+    peer.value?.off("error", errorHandler).off("open", openHandler);
```

Comment:
I wonder what does `off` do here?

And does this looks like a recursive call here?

Reply:
Just as the previous comment discussed, using error handling here is a workaround (this place is not for the main error handling logic), so this `off` make sure this handler is then removed if the peer successfully opened.